### DISCUSSION TITLE
NextJS 13 : How to get a dynamic page id in the url from a component?

### DISCUSSION THREAD

**Discussion by alamenai** [vercel/next.js#57414] [Help] [answered]
### Summary

I have a component called **Button** and inside that component, I want to make a calculation based on the ID in the route but I got blocked to find a way to access that directly from the component.

My route looks like this:

 `http://localhost:3000/75DADAS/energy-check/energy-efficiency?modernization=current&previous-page=building-details`

This is a dynamic parameter `75DADAS` which is a dynamic page in the folder structure `[id]`.

My Button.jsx:

```jsx

export default function MainButton({ investment }: { investment: IInvestment }) {
  const router = useRouter()

  const searchParams = useSearchParams()

  const modernizationStatus = searchParams.get("modernization")

  const previousPage = searchParams.get("previous-page")

  const isCurrentModernization = modernizationStatus === "current"

  const isPlannedModernization = modernizationStatus === "plan"

  const isFromBuildingDetailsPage = previousPage === "building-details"
  
 // How do you access that ID in the URL?

  const encryptedID=75DADAS

  const baseUrl = `/${encryptedId}/energy-check`

//rest of code
}
```

Currently, I can access the query parameters using `useSearchParams`,; the only way I know to access the [id] which is `75DADAS` is through `props` inside the page **energy-efficiency** but this will lead to props drilling to storing it inside a global state

**Comment by icyJoseph** [comment, marked as answer]
I think you want to use `useParams` 

- https://nextjs.org/docs/app/api-reference/functions/use-params
